src/Titon/Console/Console.hh

```diff
@@ -0,0 +1,303 @@
+<?hh // strict
+/**
+ * @copyright   2010-2015, The Titon Project
+ * @license     http://opensource.org/licenses/bsd-license.php
+ * @link        http://titon.io
+ */
+
+namespace Titon\Console;
+
+use Exception;
+use Titon\Console\Input;
+use Titon\Console\Output;
+use Titon\Console\InputDefinition\Flag;
+use ReflectionClass;
+
+/**
+ * The `Console` class bootstraps and handles Input and Output to process and
+ * run necessary commands.
+ *
+ * @package Titon\Console
+ */
+class Console {
+
+    /**
+     * A decorator banner to `brand` the application.
+     *
+     * @var string
+     */
+    protected string $banner = '';
+
+    /**
+     * The `Command` object to run
+     *
+     * @var \Titon\Console\Command|null
+     */
+    protected ?Command $command;
+
+    /**
+     * The `Input` object used to retrieve parsed parameters and commands.
+     *
+     * @var \Titon\Console\Input
+     */
+    protected Input $input;
+
+    /**
+     * The name of the application.
+     *
+     * @var string
+     */
+    protected string $name = '';
+
+    /**
+     * The `Output` object used to send response data to the user.
+     *
+     * @var \Titon\Console\Output
+     */
+    protected Output $output;
+
+    /**
+     * The version of the application.
+     *
+     * @var string
+     */
+    protected string $version = '';
+
+    /**
+     * Construct a new `Console` application.
+     */
+    public function __construct(?Input $input = null, ?Output $output = null) {
```

Comment:
No, the kernel wouldn't be a hard requirement, but another way to run the console. Think of it as a more advanced way to run the console, by utilizing middleware, and automatic handling of exit codes and termination.

If anything, it could still be ran as.

```hack
$input = new Input();
$output = new Output();

$kernel = new Kernel(new Console($input, $output), new Pipeline());
$kernel->pipe(...);
$kernel->run($input, $output); // This internally calls Console::run()
```